src/channel.rs

```diff
@@ -442,6 +442,23 @@ impl ChannelBuilder {
         let channel =
             unsafe { grpc_sys::grpc_insecure_channel_create(addr_ptr, args.args, ptr::null_mut()) };
 
+        self.connect_from_raw_channel(channel)
+    }
+
+    /// Build an insecure [`Channel`] taking over an established connection from
+    /// a file descriptor.
+    pub fn connect_from_fd(mut self, target: &str, fd: ::std::os::raw::c_int) -> Channel {
+        let args = self.prepare_connect_args();
+        let target = CString::new(target).unwrap();
+        let target_ptr = target.as_ptr();
+        let channel =
+            unsafe { grpc_sys::grpc_insecure_channel_create_from_fd(target_ptr, fd, args.args) };
+
+        self.connect_from_raw_channel(channel)
+    }
+
+    /// Build a channel from a raw underlying core grpc channel
+    pub fn connect_from_raw_channel(self, channel: *mut grpc_channel) -> Channel {
```

Comment:
Unnecessary function.

Reply:
no, this function is quite useful actually -- it allows binding custom rpc channels created by other means (e.g. third party custom transport implementations), maybe this warrants more documentation? also, might need marking as unsafe

Reply:
Can you give more details about the use case? Is it a real use case?

Reply:
Might be a 'fringe' use case, although it is 'real' for me -- you basically always need that when you intend to use anything besides 'plain' sockets for transport.